tokio/src/sync/notify.rs

```diff
@@ -277,6 +282,54 @@ impl Notify {
             waker.wake();
         }
     }
+
+    /// Notifies all waiting tasks
+    pub(crate) fn notify_waiters(&self) {
+        // Load the current state
+        let mut curr = self.state.load(SeqCst);
+
+        if let EMPTY | NOTIFIED = curr {
+            // There are no waiting tasks. In this case, no synchronization is
+            // established between `notify` and `notified().await`.
+            return;
+        }
+
+        // There are waiters, the lock must be acquired to notify.
+        let mut waiters = self.waiters.lock().unwrap();
```

Comment:
I'm not really worried about lock poisoning here. In general, I rather pretend lock poisoning isn't a thing for internal usage. This is generally how I think about Mutex across the code base. I guess we should update our internal `Mutex` shim to abort the process instead of panic then? That would be a separate PR?

Reply:
>  I guess we should update our internal `Mutex` shim to abort the process instead of panic then? That would be a separate PR?

I think it's better to update the internal mutex shim to make locking infallible by ignoring poisoned mutex errors via `into_inner`. This would result in essentially the same behavior as we get with `parking_lot` regardless of whether or not the `parking_lot` feature is enabled. Aborting the process any time a mutex is poisoned could potentially make a lot of errors, including in user code, much harder to debug.